src/api/milestone/milestone-controller.ts

```diff
@@ -67,7 +67,19 @@ export default class MilestoneController {
     }
 
     async getMilestones(_request: IRequest, h: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> {
-        const result = await this.mRepository.listAll();
+        const result = await this.mRepository.listAll().then(response => {
+            return response.sort((firstMilestone, secondMilestone) => {
+                if (firstMilestone.date > secondMilestone.date) {
+                    return 1;
+                }
+
+                if (firstMilestone.date < secondMilestone.date) {
+                    return -1;
+                }
+
+                return 0;
+            });
+        });
```

Comment:
Could you check if it's possible to use the following algorithm to perform the sorting?
Source: [StackOverflow](https://stackoverflow.com/questions/40248643/typescript-sort-by-date-not-working)

Try using the Date.getTime() method:

```
public sortByDueDate(): void {
    this.myArray.sort((a: TaskItemVO, b: TaskItemVO) => {
        return a.dueDate.getTime() - b.dueDate.getTime();

    });
}
```
^ Above throws error with undefined date so try below:

Edit
If you want to handle undefined:

```
private getTime(date?: Date) {
    return date != null ? date.getTime() : 0;
}

public sortByDueDate(): void {
    this.myArray.sort((a: TaskItemVO, b: TaskItemVO) => {
        return this.getTime(a.dueDate) - this.getTime(b.dueDate);
    });
}
```

Reply:
@ch4k4uw done.